code/Player/PropPossession.cs

```diff
@@ -99,23 +99,12 @@ private static void OnStatusChanged( Player player, PlayerStatus oldStatus )
 	}
 
 	[Event.Tick.Server]
-	private static void RechargePunchesAndCheckProp()
+	private void OnServerTick()
 	{
-		foreach ( var entity in Sandbox.Entity.All.Where( e => e.Components.Get<PropPossession>() is not null ) )
+		if ( _timeUntilRecharge )
 		{
-			var possession = entity.Components.Get<PropPossession>();
-
-			if ( !possession.Prop.IsValid() )
```

Comment:
Removing this IsValid() check causes prop breakage to go unnoticed.
Im guessing you intend to add this functionality somewhere else?
Maybe to avoid iterating over every entity every tick, a seperate list of prop possession instances could be stored serverside somewhere.

Reply:
Yeah sorry, should have mentioned I'm just trying to refactor some of this. The `linq` over every single entity seems like a code smell to me, I'm just slowly working through about how we can go about addressing that when handling the nametags.